### ISSUE TITLE
swupd: autoupdate broken (25550); can not enable

### ISSUE THREAD

**Issue by mdhorn** [clearlinux/swupd-client#621]
My system had autoupdate disabled and had fallen behind quite a bit:
```
$ sudo swupd info
Installed version: 24710
Version URL:       https://download.clearlinux.org/update/
Content URL:       https://cdn.download.clearlinux.org/update/
```

So I manually update to the latest release:
```
$ sudo swupd update
Update started.
Preparing to update from 24710 to 25550
Downloading packs...
	...35%
Extracting zstd pack for version 25470
	...100%
Error for /var/lib/swupd/pack-package-builder-from-0-to-25550.tar download: Response 200 - Failure when receiving data from the peer
Error for /var/lib/swupd/pack-x11-server-from-0-to-25550.tar download: Response 200 - Failure when receiving data from the peer
Starting download retry #1 for https://cdn.download.clearlinux.org/update//25550/pack-x11-server-from-0.tar
Starting download retry #1 for https://cdn.download.clearlinux.org/update//25550/pack-package-builder-from-0.tar
Error for /var/lib/swupd/pack-x11-server-from-0-to-25550.tar download: Response 206 - No error
Error for /var/lib/swupd/pack-package-builder-from-0-to-25550.tar download: Response 206 - No error
Starting download retry #2 for https://cdn.download.clearlinux.org/update//25550/pack-package-builder-from-0.tar
Starting download retry #2 for https://cdn.download.clearlinux.org/update//25550/pack-x11-server-from-0.tar
Error for /var/lib/swupd/pack-x11-server-from-0-to-25550.tar download: Response 206 - No error
Error for /var/lib/swupd/pack-package-builder-from-0-to-25550.tar download: Response 206 - No error
Starting download retry #3 for https://cdn.download.clearlinux.org/update//25550/pack-package-builder-from-0.tar
Starting download retry #3 for https://cdn.download.clearlinux.org/update//25550/pack-x11-server-from-0.tar
Error for /var/lib/swupd/pack-x11-server-from-0-to-25550.tar download: Response 206 - No error
Error for /var/lib/swupd/pack-package-builder-from-0-to-25550.tar download: Response 206 - No error
Starting download retry #4 for https://cdn.download.clearlinux.org/update//25550/pack-package-builder-from-0.tar
Starting download retry #4 for https://cdn.download.clearlinux.org/update//25550/pack-x11-server-from-0.tar
Error for /var/lib/swupd/pack-x11-server-from-0-to-25550.tar download: Response 206 - No error
Error for /var/lib/swupd/pack-package-builder-from-0-to-25550.tar download: Response 206 - No error
Starting download retry #5 for https://cdn.download.clearlinux.org/update//25550/pack-package-builder-from-0.tar
Starting download retry #5 for https://cdn.download.clearlinux.org/update//25550/pack-x11-server-from-0.tar
Error for /var/lib/swupd/pack-package-builder-from-0-to-25550.tar download: Response 206 - Timeout was reached
Error for /var/lib/swupd/pack-x11-server-from-0-to-25550.tar download: Response 206 - Timeout was reached

Statistics for going from version 24710 to version 25550:

    changed bundles   : 66
    new bundles       : 5
    deleted bundles   : 0

    changed files     : 297291
    new files         : 27712
    deleted files     : 40718

Starting download of remaining update content. This may take a while...
	...0%Error for /var/lib/swupd/download/.85f02a25adeec74f39d6ad7375e9833a687bf7ed5c7e6f4486a597e137035faf.tar download: Response 0 - Couldn't connect to server
	...100%
Finishing download of update content...
Starting download retry #1 for https://cdn.download.clearlinux.org/update//25440/files/85f02a25adeec74f39d6ad7375e9833a687bf7ed5c7e6f4486a597e137035faf.tar
Staging file content
Applying update
	...100%
Update was applied.
Calling post-update helper scripts.
none
rngd.service: needs a restart (a library dependency was updated)
clr_debug_fuse.service: needs a restart (a library dependency was updated)
pacdiscovery.service: needs a restart (the binary was updated)
tallow.service: needs a restart (the binary was updated)
systemd-udevd.service: needs a restart (the binary was updated)
pacrunner.service: needs a restart (the binary was updated)
systemd-journald.service: needs a restart (the binary was updated)
clr_debug_daemon.service: needs a restart (a library dependency was updated)
httpd.service: needs a restart (the binary was updated)
mcelog.service: needs a restart (the binary was updated)
systemd-resolved.service: needs a restart (the binary was updated)
systemd-timesyncd.service: needs a restart (the binary was updated)
Update took 1799.1 seconds
37706 files were not in a pack
Update successful. System updated from version 24710 to version 25550
```
Due to many service updates, I rebooted after the update:

```
 * A kernel update is available: you may wish to reboot the system.
 * Some system services need a restart.
   Run `sudo clr-service-restart -a -n` to view them.
WARNING: No system proxies configured.
See /home/mhorn/.README.txt for setup information.
$ sudo clr-service-restart -a -n
systemd-networkd.service: needs a restart (the binary was updated)
/usr/bin/systemctl --no-ask-password try-restart systemd-networkd.service
docker.service: needs a restart (the binary was updated)
/usr/bin/systemctl --no-ask-password try-restart docker.service
polkit.service: needs a restart (a library dependency was updated)
/usr/bin/systemctl --no-ask-password try-restart polkit.service
bluetooth.service: needs a restart (a library dependency was updated)
/usr/bin/systemctl --no-ask-password try-restart bluetooth.service
accounts-daemon.service: needs a restart (the binary was updated)
/usr/bin/systemctl --no-ask-password try-restart accounts-daemon.service
wpa_supplicant.service: needs a restart (a library dependency was updated)
/usr/bin/systemctl --no-ask-password try-restart wpa_supplicant.service
ModemManager.service: needs a restart (the binary was updated)
/usr/bin/systemctl --no-ask-password try-restart ModemManager.service
fwupd.service: needs a restart (the binary was updated)
/usr/bin/systemctl --no-ask-password try-restart fwupd.service
colord.service: needs a restart (a library dependency was updated)
/usr/bin/systemctl --no-ask-password try-restart colord.service
gdm.service: needs a restart (the binary was updated)
/usr/bin/systemctl --no-ask-password try-restart gdm.service
upower.service: needs a restart (a library dependency was updated)
/usr/bin/systemctl --no-ask-password try-restart upower.service
dbus.service: needs a restart (a library dependency was updated)
/usr/bin/systemctl --no-ask-password try-restart dbus.service
systemd-logind.service: needs a restart (the binary was updated)
/usr/bin/systemctl --no-ask-password try-restart systemd-logind.service

```

After the reboot I attempt to enable autoupdate:
```
$ sudo swupd autoupdate
Disabled
$ sudo swupd autoupdate --enable
Running systemctl to enable updates
Removed /etc/systemd/system/swupd-update.service.
Removed /etc/systemd/system/swupd-update.timer.
Failed to start swupd-update.timer: Unit swupd-update.timer not found.
```
Tried a second time:
```
$ sudo swupd autoupdate --enable
Running systemctl to enable updates
Unit swupd-update.service does not exist, proceeding anyway.
Unit swupd-update.timer does not exist, proceeding anyway.
Failed to start swupd-update.timer: Unit swupd-update.timer not found.
$ sudo swupd autoupdate
Failed to get unit file state for swupd-update.service: No such file or directory
Disabled
```
Attempted a Verify:
```
$ sudo swupd verify
Verifying version 25550
Verifying files
	...100%
Inspected 320900 files
Verify successful
```

Verify is clean, tried again:
```
$ sudo swupd autoupdate --enable
Running systemctl to enable updates
Unit swupd-update.service does not exist, proceeding anyway.
Unit swupd-update.timer does not exist, proceeding anyway.
Failed to start swupd-update.timer: Unit swupd-update.timer not found.
$ sudo swupd autoupdate
Failed to get unit file state for swupd-update.service: No such file or directory
Disabled
```

Okay, Verify Fix:
```
$ sudo swupd verify --fix
Verifying version 25550
Verifying files
	...100%
Starting download of remaining update content. This may take a while...
	...100%
Finishing download of update content...
Adding any missing files

Fixing modified files
	...100%
Inspected 320900 files
  0 files were missing
  0 files found which should be deleted
Calling post-update helper scripts.
none
Fix successful
$ sudo swupd autoupdate --enable
Running systemctl to enable updates
Unit swupd-update.service does not exist, proceeding anyway.
Unit swupd-update.timer does not exist, proceeding anyway.
Failed to start swupd-update.timer: Unit swupd-update.timer not found.
```

Still not able to enable auto-update.

Next steps?

**Comment by matthewrsj** [comment]
I can reproduce on the latest version

**Comment by otaviobp** [comment]
Fixed on Release 3.18.1. Waiting for more feedback before closing this issue. Thanks for the report
